### PR TITLE
test: move flaky tests to quarantine

### PR DESCRIPTION
**Pull request by wes-turner**
Creates a new nightly-quarantine for flaky nightly tests.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by wes-turner** [comment]
When should quarantined tests be run? The CircleCI config is hard to parse, but currently it looks like they're only a part of `test-e2e-aws` which is run... where?

**Comment by ioga** [comment]
> When should quarantined tests be run? The CircleCI config is hard to parse, but currently it looks like they're only a part of `test-e2e-aws` which is run... where?

`test-e2e-quarantined` seems to run everywhere except for nightly: on master,  upstream branches, and forks

**Comment by wes-turner** [comment]
Thanks, @ioga. That seems like the proper home for both these new quarantined tests. I'd gotten confused by `test-e2e-gpu-mmdetection-quarantine`. I'll move that one to the quarantined workflow, too.

**Comment by wes-turner** [comment]
Well, now reading through the "CI triage runbook", I no longer think that this is the intended place for the quarantine tests. Instead, I think the runbook says, they should run in the same workflow in a parallel, "quarantine" job. Which also means that the change isn't testable as written, since the workflow is run as part of "nightly" and not from a branch.

File: .circleci/config.yml [hunk 1]
```diff
@@ -3355,6 +3355,14 @@ workflows:
               parallelism: [2]
               cluster-id-prefix: ["nightly"]
               mark: ["nightly"]
+      - test-e2e-aws:
```
Comment (MikhailKardash) [inline .circleci/config.yml:3358]: Right now this is only under `nightly` run, should we have a copy of this under `test-e2e-longrunning`? See: https://github.com/determined-ai/determined/pull/7520/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R3002
I guess it isn't required for PRs, but in my opinion `nightly` is not very visible for devs.
